pkg/registry/apis/query/query.go

```diff
@@ -122,23 +121,32 @@ func (r *queryREST) Connect(connectCtx context.Context, name string, _ runtime.O
 		// Parses the request and splits it into multiple sub queries (if necessary)
 		req, err := b.parser.parseRequest(ctx, raw)
 		if err != nil {
-			reason := metav1.StatusReasonInvalid
-			message := err.Error()
+			var refError ErrorWithRefID
+			statusCode := http.StatusBadRequest
+			message := err
+			refID := "A"
```

Comment:
nit: i wonder if we could default to `""` (empty string).

Reply:
Good call :+1: